packages/tenderly-hardhat/src/tasks/push.ts

```diff
@@ -1,51 +1,14 @@
-import { HardhatPluginError } from "hardhat/plugins";
 import { task } from "hardhat/config";
-import { TenderlyService } from "tenderly";
 
-import { logger } from "../utils/logger";
-import { PLUGIN_NAME } from "../constants";
-import { newCompilerConfig } from "../utils/util";
-import { extractContractData } from "./common";
+import { HardhatRuntimeEnvironment } from "hardhat/types";
 
-const tenderlyService = new TenderlyService(PLUGIN_NAME);
-
-task("tenderly:push", "Privately pushes contracts to Tenderly")
+task("tenderly:push", "Verifies contracts on Tenderly based on the configuration in hardhat.config.js.")
   .addOptionalVariadicPositionalParam(
     "contracts",
     "Addresses and names of contracts that will be verified formatted ContractName=Address"
   )
   .setAction(pushContracts);
 
-async function pushContracts({ contracts }: any, { config, hardhatArguments, run }: any) {
-  logger.info("Private verification hardhat task has been invoked.");
-
-  if (contracts === undefined) {
-    throw new HardhatPluginError(PLUGIN_NAME, `At least one contract must be provided (ContractName=Address)`);
-  }
-
-  if (config.tenderly.project === undefined) {
-    throw new HardhatPluginError(
-      PLUGIN_NAME,
-      `Please provide the project field in the tenderly object in hardhat.config.js`
-    );
-  }
-
-  if (config.tenderly.username === undefined) {
-    throw new HardhatPluginError(
-      PLUGIN_NAME,
-      `Please provide the username field in the tenderly object in hardhat.config.js`
-    );
-  }
-
-  const requestContracts = await extractContractData(contracts, hardhatArguments.network, config, run);
-  const solcConfig = newCompilerConfig(config);
-
-  await tenderlyService.pushContracts(
-    {
-      config: solcConfig,
-      contracts: requestContracts,
-    },
-    config.tenderly.project,
-    config.tenderly.username
-  );
+async function pushContracts({ contracts }: any, hre: HardhatRuntimeEnvironment) {
+  await hre.run("tenderly:verify", { contracts });
```

Comment:
I have two things to say about this line. First thing is a big miss on my side that I have to explain, the other is trying to answer to your question.

The miss on my side that I realised yesterday:  
Till now, for tasks, the configuration wasn't important. 
If you called `tenderly:verify`, it was a public verification. 
If you called `tenderly:push`, it was a private verification.

Let's say someone downloads this version of the plugin and wants to privately verify their contracts via `tenderly:push` task. By current implementation, if you have your configuration set for public verification, `tenderly:push` will verify the contracts publicly, which is not acceptable since it is not expected behaviour.

What I was trying to achieve with this change is to not be ambiguous. As a user, If there is a config that says what type of verification I want, then the config should be the only place I can configure that. I wouldn't like to set my config for public verification and then call `tenderly:push` only to find out my contracts are privately verified because "that's how this task works", but If I go to my `deploy-script.js` and call `hre.tenderly.push(...contracts)` then the verification type will depend on my configuration.
So in order to not be ambiguous, we have to options:
- delete the config, `push` is for private, `verify` is for public, `verify` or `push` or `forkVerify` is for fork verification.
- keep the config, `verify` works based on the config.

I think the latter works best since it will require much less work and it will feel more natural to have one method that works based on your configuration since that's the current implementation and behaviour.
It's only that tasks didn't work this way, which is what I am proposing to change to have a unified approach to verification.

In order to fix the miss that I made (disturbing expected behaviour of verifying contracts privately),
I propose to put a prompt on both tasks since they both have unexpected behaviour, it's just that one can do far more damage than the other one.
The prompt would say something like `hey, this was public/private verification before, but now we have changed something and your configuration says otherwise, would you like to continue?` (I know this is a really bad message, but we can say something else 😄). 

Finally, to answer your question on how can some contracts be verified publicly and some privately.
Users has to set the configuration, then call the verify task, then change the configuration, then call the verify task.
Regarding `hre.tenderly.verify` verification. User can change the `hre.config.tenderly.privateVerification` flag in the middle of the code if there is a need to verify part of the contracts privately and the other part publicly. We can put this in readme.md for better visibility.
 
 Sorry for the long post, here's a potato. 🥔

Reply:
Ok, this seems pretty reasonable and I totally agree with your logic here. Just wondering how this is going to operate for automatic verification. I would assume it should be the same both for fork and normal verification, just wanted your confirmation.

Reply:
As we agreed, we will leave `tenderly:push` for private verification ONLY. We don't want to change this functionality since it is risky. 
There will be a deprecation log when using `tenderly:push` task, redirecting user to `tenderly:verify`.

Reply:
I have put a deprecation warning on the `tenderly:push` task. 
Also, I changed the functionality back to expected. When the user wants to verify contracts via `tenderly:push`, it will be a private verification.